client/trino-client/src/main/java/io/trino/client/ClientSession.java

```diff
@@ -307,8 +307,12 @@ private Builder(ClientSession clientSession)
             traceToken = clientSession.getTraceToken();
             clientTags = clientSession.getClientTags();
             clientInfo = clientSession.getClientInfo();
-            catalog = clientSession.getCatalog();
-            schema = clientSession.getSchema();
+            if (clientSession.getCatalog().isPresent()) {
+                catalog = clientSession.getCatalog().get();
```

Comment:
can we just assign `catalog = clientSession.getCatalog().orElse(null)`, the same for schema? latter in the code we are converting that to the `Optional` with the `.ofNullable` call

Reply:
Done